File: examples/javascript/automation-logs-client.mjs
const BASE_URL = process.env.BASE_URL || 'http://localhost:3001';

async function request(method, path, payload) {
  const response = await fetch(`${BASE_URL}${path}`, {
    method,
    headers: payload ? { 'Content-Type': 'application/json' } : {},
    body: payload ? JSON.stringify(payload) : undefined,
  });

  if (!response.ok) {
    throw new Error(`HTTP ${response.status}: ${await response.text()}`);
  }

  return response.json();
}

export async function getAutomation(automationId) {
  try {
    return await request('GET', `/automations/${automationId}`);
  } catch (error) {
    if (String(error.message).startsWith('HTTP 404:')) {
      return null;
    }
    throw error;
  }
}

export async function createAutomation({ id, name, type, version, manualExecutionEffortMinutes }) {
  return request('POST', '/automations', {
    id,
    name,
    type,
    version,
    manualExecutionEffortMinutes,
  });
}

export async function ensureAutomation({ id, name, type, version, manualExecutionEffortMinutes }) {
  const existing = await getAutomation(id);
  if (existing) {
    return existing;
  }

  return createAutomation({ id, name, type, version, manualExecutionEffortMinutes });
}

export async function sendExecution(automationId, execution) {
  return request('POST', '/executions', {
    automation_id: automationId,
    ...execution,
  });
}

const automation = await ensureAutomation({
  id: process.env.AUTOMATION_ID || 'replace-with-approved-automation-id',
  name: 'Backup Firewall',
  type: 'javascript',
  version: '1.0.0',
  manualExecutionEffortMinutes: 30,
});

console.log('Using automation:', automation.id);

const executionResult = await sendExecution(automation.id, {
  execution_uuid: process.env.EXECUTION_UUID || 'a7b1f570-85fb-4b4d-bb63-19ce0d75dfe4',
  trigger_type: 'schedule',
  status: 'success',
  started_at: '2026-03-31T19:00:00.000Z',
  finished_at: '2026-03-31T19:01:23.000Z',
  duration_ms: 83000,
  host: 'runner-01',
  runtime: 'node20',
  error_code: null,
  error_message: null,
});

console.log('Execution response:', executionResult);

File: examples/python/automation_logs_client.py
#!/usr/bin/env python3
import json
import os
import urllib.parse
import urllib.request
import urllib.error

BASE_URL = os.getenv("BASE_URL", "http://localhost:3001")


def _request(method: str, path: str, payload: dict | None = None) -> dict:
    data = None
    headers = {}
    if payload is not None:
        data = json.dumps(payload).encode("utf-8")
        headers["Content-Type"] = "application/json"

    req = urllib.request.Request(
        f"{BASE_URL}{path}",
        data=data,
        method=method,
        headers=headers,
    )
    try:
        with urllib.request.urlopen(req) as response:
            return json.loads(response.read().decode("utf-8"))
    except urllib.error.HTTPError as error:
        body = error.read().decode("utf-8")
        raise RuntimeError(f"HTTP {error.code}: {body}") from error


def get_automation(automation_id: str) -> dict | None:
    try:
        return _request("GET", f"/automations/{automation_id}")
    except RuntimeError as error:
        if "HTTP 404:" in str(error):
            return None
        raise


def create_automation(
    automation_id: str,
    name: str,
    automation_type: str,
    version: str,
    manual_minutes: int | None = None,
) -> dict:
    payload = {
        "id": automation_id,
        "name": name,
        "type": automation_type,
        "version": version,
    }
    if manual_minutes is not None:
        payload["manualExecutionEffortMinutes"] = manual_minutes
    return _request("POST", "/automations", payload)


def ensure_automation(
    automation_id: str,
    name: str,
    automation_type: str,
    version: str,
    manual_minutes: int | None = None,
) -> dict:
    existing = get_automation(automation_id)
    if existing:
        return existing

    return create_automation(
        automation_id,
        name,
        automation_type,
        version,
        manual_minutes,
    )


def send_execution(automation_id: str, execution: dict) -> dict:
    return _request(
        "POST",
        "/executions",
        {
            "automation_id": automation_id,
            **execution,
        },
    )


if __name__ == "__main__":
    automation_id = os.getenv(
        "AUTOMATION_ID", "replace-with-approved-automation-id"
    )
    automation = ensure_automation(
        automation_id,
        "Backup Firewall",
        "python",
        "1.0.0",
        30,
    )
    print("Using automation:", automation["id"])

    result = send_execution(
        automation["id"],
        {
            "execution_uuid": os.getenv(
                "EXECUTION_UUID", "a7b1f570-85fb-4b4d-bb63-19ce0d75dfe4"
            ),
            "trigger_type": "schedule",
            "status": "success",
            "started_at": "2026-03-31T19:00:00.000Z",
            "finished_at": "2026-03-31T19:01:23.000Z",
            "duration_ms": 83000,
            "host": "runner-01",
            "runtime": "python3.11",
            "error_code": None,
            "error_message": None,
        },
    )
    print("Execution response:", json.dumps(result, indent=2))
